### PR TITLE
feat: add blocks/Form

### PR DESCRIPTION
**Pull request by qradle-yndx**
* add form-block
* add validator schemas for YandexForm and HubspotForm

### CODE REVIEW

File: styles/variables.scss [hunk 1]
```diff
@@ -53,3 +53,5 @@ $gridGutterMobile: 8px;
 $contentWidth: calc(#{$newContentWidth} + #{$gridGutter} * 2 + #{$gridContainerPadding} * 2);
 
 $gridContainerMargin: 16px;
+
+$largeBorderRadius: 32px;
```
Comment (gorgeousvlad) [inline styles/variables.scss:57]: I think we don't need to add this variable here, just use it in `Form.scss`

File: src/blocks/Form/__stories__/data.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "disableCompress": true
+      }
+    }
+  },
+  "black": {
+    "content": {
+      "textContent": {
+        "theme": "dark",
+        "title": "Black form background color",
+        "text": "<p>We&nbsp;are all storytellers. Stories are a&nbsp;powerful way to&nbsp;communicate ideas and share information. The right story can lead to&nbsp;a&nbsp;better understanding of&nbsp;a&nbsp;situation, make us&nbsp;laugh, or&nbsp;even inspire us&nbsp;to&nbsp;do&nbsp;something in&nbsp;the future.</p>"
```
Comment (gorgeousvlad) [inline src/blocks/Form/__stories__/data.json:38]: we use `Lorem ipsum...` for example texts it is not necessary, but good for unifiacation

File: src/blocks/Form/Form.scss [hunk 3]
```diff
@@ -0,0 +1,230 @@
+@import '../../../styles/mixins.scss';
+@import '../../../styles/variables.scss';
+
+$block: '.#{$ns}form-block';
+
+$textPadding: 10px;
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.scss:6]: do you mean using $indentXXXS (8px) or $indentXXS (12px) from variables?

**Review by aeksandla — APPROVED** [review]

File: src/blocks/Form/Form.scss [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            }
+        }
+    }
+
+    @media (max-width: map-get($gridBreakpoints, 'lg')) and (min-width: map-get($gridBreakpoints, 'md')) {
+        &__row {
+            flex-direction: column;
+        }
+
+        &_with-background,
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.scss:138]: this classes does make sense inside @media queries

File: src/blocks/Form/Form.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.scss:6]: this is indent from yandex/hubspot form

File: src/blocks/Form/Form.scss [hunk 5]
```diff
@@ -0,0 +1,230 @@
+@import '../../../styles/mixins.scss';
+@import '../../../styles/variables.scss';
+
+$block: '.#{$ns}form-block';
+
+$textPadding: 10px;
+$maxLgWidth: 609px;
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.scss]: changed to maxLargeWidth

File: src/blocks/Form/Form.scss [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                }
+
+                #{$root}__content-wrapper {
+                    text-align: center;
+                    padding-bottom: $indentM;
+                }
+            }
+        }
+
+        &_with-background &__row {
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.scss]: removed

File: src/blocks/Form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        textContent,
+        image,
+        direction = FormBlockDirection.Center,
+        backgroundColor,
+    } = props;
+    const [contentLoaded, setContentLoaded] = useState(false);
+    const isMobile = useContext(MobileContext);
+
+    const withBackground = Boolean(backgroundColor) || Boolean(image);
+    const colSizes = {[GridColumnSize.Lg]: 6, [GridColumnSize.All]: 12};
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.tsx]: moved const outside of FormBlock component

File: src/blocks/Form/Form.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    const paddingBottom = Boolean(backgroundColor) && !image ? 'l' : 'm';
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.tsx]: added a comment

File: src/blocks/Form/Form.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+        return undefined;
+    }, [formData]);
+
+    useEffect(() => {
+        if (formType === FormBlockDataTypes.HUBSPOT_INLINE) {
+            onContentLoad();
+        }
+    }, [formType, onContentLoad]);
+
+    const renderForm = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.tsx]: moved to separate InnerForm component

File: src/blocks/Form/Form.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+                            </div>
+                        )}
+                    </Col>
+
+                    <Col sizes={colSizes} className={b('form-col')}>
+                        <div className={b('form-wrapper')}>
+                            <div
+                                className={b('full-form', {
+                                    hidden: !contentLoaded,
+                                    'without-title': !textContent,
```
[the remark this answers is not in the document]
Comment (qradle-yndx) [reply src/blocks/Form/Form.tsx]: removed

File: src/blocks/Form/__stories__/data.json (hunk 2 above)
Comment (qradle-yndx) [reply src/blocks/Form/__stories__/data.json]: changed

**Review by yuberdysheva — CHANGES_REQUESTED** [review]

File: src/blocks/Form/Form.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    const isMobile = useContext(MobileContext);
+
+    const hasImage = background && (background.src || background.desktop);
+    const paddingBottom = background && background.style?.backgroundColor && !hasImage ? 'l' : 'm'; // bigger padding for case with background color and no image
+    const onContentLoad = useCallback(() => {
+        setContentLoaded(true);
+    }, []);
+
+    const formType = useMemo(() => {
+        if (isYandexDataForm(formData)) return FormBlockDataTypes.YANDEX;
```
Comment (gorgeousvlad) [inline src/blocks/Form/Form.tsx:36]: we don't use one-line if's notation

File: src/components/YandexForm/__stories__/data.json [hunk 12]
```diff
@@ -1,7 +1,8 @@
 {
   "default": {
     "content": {
-      "id": "10034371.a94dac9d3bd19f65c9616db0297348ad8d063db0"
+      "id": "61a4e639d4d24e0dbba36f5c",
+      "customFormSection": "cloud"
```
Comment (gorgeousvlad) [inline src/components/YandexForm/__stories__/data.json:5]: is this form somehow related to Yandex Cloud?

Comment (yuberdysheva) [reply src/components/YandexForm/__stories__/data.json:5]: no, i created it as a regular user without any specific text from Yandex Cloud

File: src/blocks/Form/__stories__/Form.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+                    ...args,
+                    formData: {hubspot: data.formData.hubspot}, //hubspot form
+                },
+            ],
+        }}
+    />
+);
+
+export const Default = DefaultTemplate.bind({});
+export const ContentForm = DefaultTemplate.bind({});
```
Comment (yuberdysheva) [inline src/blocks/Form/__stories__/Form.stories.tsx:39]: Content
```suggestion
export const Content Direction = DefaultTemplate.bind({});
```

Lets add here the `form-content`

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,6 +1,13 @@
 # Changelog
 
 
+## [4.31.1](https://github.com/gravity-ui/page-constructor/compare/v4.31.0...v4.31.1) (2023-10-30)
```
Comment (gorgeousvlad) [inline CHANGELOG.md:4]: changelog is the generated by release robot, you don't need to do it manually

File: package.json [hunk 15]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@gravity-ui/page-constructor",
-  "version": "4.31.0",
+  "version": "4.31.1-alpha.0",
```
Comment (gorgeousvlad) [inline package.json:3]: remove pls

File: src/blocks/Form/Form.tsx [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+    const [contentLoaded, setContentLoaded] = useState(false);
+    const isMobile = useContext(MobileContext);
+
+    const hasImage = background && (background.src || background.desktop);
+    const paddingBottom = background && background.style?.backgroundColor && !hasImage ? 'l' : 'm'; // bigger padding for case with background color and no image
+    const onContentLoad = useCallback(() => {
+        setContentLoaded(true);
+    }, []);
+
+    const formType = useMemo(() => {
```
Comment (gorgeousvlad) [inline src/blocks/Form/Form.tsx:35]: useMemo is not necessary

**Review by gorgeousvlad — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 14 above)
Comment (qradle-yndx) [reply CHANGELOG.md]: I did not. It was my messy rebase main branch. fixed

File: src/models/constructor-items/blocks.ts [hunk 17]
```diff
@@ -376,6 +378,35 @@ export interface ShareBlockProps {
     title?: string;
 }
 
+export enum FormBlockDataTypes {
+    YANDEX = 'yandex',
+    HUBSPOT_INLINE = 'hubspot-inline',
```
Comment (yuberdysheva) [inline src/models/constructor-items/blocks.ts:383]: can we do that if this is not major?
```suggestion
    HUBSPOT = 'hubspot',
```

File: src/blocks/Form/schema.ts [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+                    {
+                        type: 'object',
+                        optionName: 'yandex',
+                        properties: {
+                            yandex: YandexFormProps,
+                        },
+                    },
+                    {
+                        type: 'object',
+                        optionName: 'hubspot-inline',
```
Comment (yuberdysheva) [inline src/blocks/Form/schema.ts:31]:
```suggestion
                        optionName: 'hubspot',
```

**Review by yuberdysheva — APPROVED** [review]

**Review by gorgeousvlad — APPROVED** [review]
